mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/ContainerEventHandlerServiceConfigurator.java

```diff
@@ -35,45 +37,54 @@
 import org.wildfly.clustering.service.AsyncServiceConfigurator;
 import org.wildfly.clustering.service.FunctionalService;
 import org.wildfly.clustering.service.ServiceConfigurator;
-import org.wildfly.clustering.service.SimpleServiceNameProvider;
 
 /**
  * Service starting mod_cluster service.
  *
  * @author Jean-Frederic Clere
  * @author Radoslav Husar
  */
-public class ContainerEventHandlerServiceConfigurator extends SimpleServiceNameProvider implements ServiceConfigurator, Supplier<ModClusterService>, Consumer<ModClusterService> {
+public class ContainerEventHandlerServiceConfigurator implements ServiceConfigurator, Supplier<ModClusterService>, Consumer<ModClusterService> {
 
-    public static final ServiceName SERVICE_NAME = ModClusterConfigResourceDefinition.MOD_CLUSTER_CAPABILITY.getCapabilityServiceName();
-    public static final ServiceName CONFIG_SERVICE_NAME = SERVICE_NAME.append("config");
+    private final String proxyName;
+    private final LoadBalanceFactorProvider factorProvider;
+    private volatile Supplier<ModClusterConfiguration> configuration;
 
-    private final LoadBalanceFactorProvider load;
+    ContainerEventHandlerServiceConfigurator(String proxyName, LoadBalanceFactorProvider factorProvider) {
+        this.proxyName = proxyName;
+        this.factorProvider = factorProvider;
+    }
 
-    private volatile Supplier<ModClusterConfiguration> configuration;
+    @Override
+    public ServiceName getServiceName() {
+        return getServiceName(proxyName);
+    }
 
-    ContainerEventHandlerServiceConfigurator(LoadBalanceFactorProvider factorProvider) {
-        super(SERVICE_NAME);
-        this.load = factorProvider;
+    public static ServiceName getServiceName(String proxyName) {
+        return ProxyConfigurationResourceDefinition.Capability.SERVICE.getDefinition().fromBaseCapability(proxyName).getCapabilityServiceName();
     }
 
+
     @Override
     public ServiceBuilder<?> build(ServiceTarget target) {
         ServiceBuilder<?> builder = new AsyncServiceConfigurator(this.getServiceName()).build(target);
-        // Temporary alias, in case anyone is using old service name
-        Consumer<ModClusterService> modClusterService = builder.addAliases(ServiceName.JBOSS.append(ModClusterExtension.SUBSYSTEM_NAME)).provides(this.getServiceName());
-        this.configuration = builder.requires(ContainerEventHandlerServiceConfigurator.CONFIG_SERVICE_NAME);
+        Consumer<ModClusterService> modClusterService = builder.provides(this.getServiceName());
+        this.configuration = builder.requires(ProxyConfigurationServiceConfigurator.getServiceName(proxyName));
```

Comment:
Consumer<ModClusterService> modClusterService = this.configuration.register(builder).provides(this.getServiceName());